src/Parser.y

```diff
@@ -543,7 +575,9 @@ addIdToSymTable mi d@(c, gId@(G.Id tk@(T.Token {T.aToken=at, T.cleanedString=idN
       else if category == ST.Function
       then RWS.tell $ [ST.SemanticError ("Name " ++ show tk ++ " conflicts with a function") tk]
       else if category == ST.RefParam || category == ST.ValueParam
-      then RWS.tell $ [ST.SemanticError ("Name "++ show tk ++ " conflicts with a formal param") tk]
+      then RWS.tell $ [ST.SemanticError ("Name " ++ show tk ++ " conflicts with a formal param") tk]
+      else if elem (T.cleanedString tk) iterVars
```

Comment:
better to use `elem` as an "operator"
```haskell
`elem`
```

Reply:
changed in both uses that arised from this PR